src/main/java/xbot/common/math/kinematics/DeadwheelWheelSpeeds.java

```diff
@@ -0,0 +1,181 @@
+package xbot.common.math.kinematics;
+
+import static edu.wpi.first.units.Units.MetersPerSecond;
+
+import edu.wpi.first.units.measure.LinearVelocity;
+
+/** Represents the wheel speeds for deadwheels. */
+public class DeadwheelWheelSpeeds {
+    /** Speed of the left side of the robot. */
+    public double leftMetersPerSecond;
+
+    /** Speed of the right side of the robot. */
+    public double rightMetersPerSecond;
+
+    /** Speed of the right side of the robot. */
+    public double frontMetersPerSecond;
+
+    /** Speed of the right side of the robot. */
+    public double rearMetersPerSecond;
+
+    /** Constructs a DeadwheelWheelSpeeds with zeros for left and right speeds. */
+    public DeadwheelWheelSpeeds() {
+    }
+
+    /**
+     * Constructs a DifferentialDriveWheelSpeeds.
+     *
+     * @param leftMetersPerSecond  The left speed.
+     * @param rightMetersPerSecond The right speed.
+     * @param frontMetersPerSecond The front speed.
+     * @param rearMetersPerSecond  The rear speed.
+     */
+    public DeadwheelWheelSpeeds(double leftMetersPerSecond, double rightMetersPerSecond, double frontMetersPerSecond,
+            double rearMetersPerSecond) {
+        this.leftMetersPerSecond = leftMetersPerSecond;
+        this.rightMetersPerSecond = rightMetersPerSecond;
+        this.frontMetersPerSecond = frontMetersPerSecond;
+        this.rearMetersPerSecond = rearMetersPerSecond;
+    }
+
+    /**
+     * Renormalizes the wheel speeds if any either side is above the specified
+     * maximum.
+     *
+     * <p>
+     * Sometimes, after inverse kinematics, the requested speed from one or more
+     * wheels may be
+     * above the max attainable speed for the driving motor on that wheel. To fix
+     * this issue, one can
+     * reduce all the wheel speeds to make sure that all requested module speeds are
+     * at-or-below the
+     * absolute threshold, while maintaining the ratio of speeds between wheels.
+     *
+     * @param attainableMaxSpeedMetersPerSecond The absolute max speed that a wheel
+     *                                          can reach.
+     */
+    public void desaturate(double attainableMaxSpeedMetersPerSecond) {
```

Comment:
:star: Given that this is telemetry, rather than drive, it doesn't seem like we need this function?